pkg/sql/logictest/testdata/logic_test/schema_repair

```diff
@@ -0,0 +1,189 @@
+    )
+SELECT
+  descriptor
+FROM
+  descs
+WHERE
+  id = $corrupt_id;
+
+# Delete our corrupt descriptor
+query B
+SELECT * FROM ROWS FROM (crdb_internal.unsafe_delete_descriptor($corrupt_id));
+----
+true
+
+# Force delete the descriptor
+query B
+SELECT * FROM ROWS FROM (crdb_internal.unsafe_delete_descriptor($t_id));
+----
+true
+
+# Corrupt the descriptor with fake ID's
+let $json_t_corrupt
+SELECT CAST(replace('$json_corrupt','"name": "corruptdesc",', '') AS JSONB)
+
+# Inject our corrupt descriptor with the wrong ID
+statement ok
+SELECT * FROM crdb_internal.unsafe_upsert_descriptor($corrupt_id, crdb_internal.json_to_pb( 'cockroach.sql.sqlbase.Descriptor','$json_t_corrupt'), true)
+
+
+query I
+SELECT count(*) FROM crdb_internal.lost_descriptors_with_data WHERE descid = $t_id;
+----
+1
+
+query I
+SELECT count(*) FROM crdb_internal.lost_descriptors_with_data WHERE descid != $t_id
+----
+0
+
+statement ok
+SELECT * FROM crdb_internal.unsafe_upsert_descriptor($t_id, crdb_internal.json_to_pb( 'cockroach.sql.sqlbase.Descriptor','$json_t'))
+
+# Recover the corrupted descriptor
+statement ok
+SELECT * FROM crdb_internal.unsafe_upsert_descriptor($corrupt_id, crdb_internal.json_to_pb( 'cockroach.sql.sqlbase.Descriptor','$json_corrupt'), true)
+
+statement ok
+SELECT * FROM corruptdesc;
+
+subtest force-delete-data
+
+statement ok
+CREATE TABLE forcedeletemydata (v int)
+
+statement ok
+INSERT INTO forcedeletemydata VALUES(5)
+
+statement ok
+INSERT INTO forcedeletemydata VALUES(7)
+
+query I
+SELECT * FROM forcedeletemydata ORDER BY v ASC
+----
+5
+7
+
+# Blocked since ID was not used
+statement error descriptor id was never used
+select * from crdb_internal.force_delete_table_data(6666)
+
+
+let $t_id
+select id from system.namespace where name='forcedeletemydata'
+
+# Descriptor exists so operation is blocked
+statement error descriptor still exists force deletion is blocked
+select * from crdb_internal.force_delete_table_data($t_id)
+
+query I
+SELECT * FROM forcedeletemydata ORDER BY v ASC
+----
+5
+7
+
+let $parentID
+select pid from system.namespace2 as n(pid,psid,name,id) where id=$t_id;
+
+let $parentSchemaID
+select psid from system.namespace2 as n(pid,psid,name,id)  where id=$t_id;
+
+let $json
+WITH descs AS (
+                SELECT id,
+                       crdb_internal.pb_to_json(
+                        'cockroach.sql.sqlbase.Descriptor',
+                        descriptor
+                       ) AS descriptor
+                  FROM system.descriptor
+             )
+select descriptor from descs where id=$t_id;
+
+
+# Force delete the descriptor
+query B
+select * from crdb_internal.unsafe_delete_descriptor($t_id);
+----
+true
+
+query B
+select * from crdb_internal.force_delete_table_data($t_id)
+----
+true
+
+statement ok
+select * from crdb_internal.unsafe_upsert_descriptor($t_id, crdb_internal.json_to_pb( 'cockroach.sql.sqlbase.Descriptor','$json'))
+
+query I
+SELECT * FROM forcedeletemydata ORDER BY v ASC
+----
```

Comment:
For some reason all these logic tests end up in the first of your two commits.

Reply:
Yeah, not sure why that happened this was done via backport. It should have been spread between the two commits and I saw signs of that happening with the merge conflicts. I think it's harmless, and not worth digging into and I'll merge once CI comes back.

Reply:
I suppose it only matters if one would want to revert the second of the two commits, but not the first.